api/src/main/java/com/epam/pipeline/controller/metadata/CategoricalAttributeController.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2017-2020 EPAM Systems, Inc. (https://www.epam.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.controller.metadata;
+
+import com.epam.pipeline.controller.AbstractRestController;
+import com.epam.pipeline.controller.Result;
+import com.epam.pipeline.manager.metadata.CategoricalAttributeApiService;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.util.List;
+import java.util.Map;
+
+@Controller
+@Api(value = "CategoricalAttributes")
+@RequiredArgsConstructor
+@RequestMapping(value = "/categoricalAttribute")
+public class CategoricalAttributeController extends AbstractRestController {
+
+    private final CategoricalAttributeApiService categoricalAttributeApiService;
```

Comment:
Let's introduce a separate class `CategoricalAttribute` and use it instead of Map<String, List<String, String>>. It will increase readability of the code and may be useful if decide to introduce more complex ACL permissions.

Reply:
Done, using `List<CategoricalAttribute>` instead of `Map<String, List<String>>`